src/lsm/composite_key.zig

```diff
@@ -4,67 +4,73 @@ const math = std.math;
 
 const stdx = @import("../stdx.zig");
 
-pub fn CompositeKey(comptime Field: type) type {
-    assert(Field == u128 or Field == u64);
+pub fn CompositeKeyType(comptime Field: type) type {
+    // The type if zeroed padding is needed.
+    const pad = switch (Field) {
+        u128 => @as(u64, 0),
+        u64 => {},
+        else => @compileError("invalid Field for CompositeKey: " ++ @typeName(Field)),
+    };
+    const Pad = @TypeOf(pad);
+
+    // The composite key's backing integer type.
+    const Int = std.meta.Int(
+        .unsigned,
+        @bitSizeOf(u64) + @bitSizeOf(Field) + @bitSizeOf(Pad),
+    );
```

Comment:
As we only support u128 and u64 fields, maybe this will be more readable as an explicit switch with hard-coded `u128` and `u256` arms? Works either way for me.